src/main/java/ulcrs/controllers/CourseController.java

```diff
@@ -0,0 +1,51 @@
+package ulcrs.controllers;
+
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import spark.Request;
+import spark.Response;
+import spark.RouteGroup;
+import spark.Spark;
+import ulcrs.models.course.Course;
+import ulcrs.models.course.CourseIntensity;
+import ulcrs.models.course.CourseRequirements;
+import ulcrs.models.tutor.Tutor;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.List;
+
+public class CourseController extends BaseController {
+
+    @Override
+    public RouteGroup routes() {
+        return () -> {
+            Spark.before("/*", (request, response) -> log.info("endpoint: " + request.pathInfo()));
+            Spark.get("/", this::getCourseList, gson::toJson);
+            Spark.get("/:id", this::getCourse, gson::toJson);
+        };
+    }
+
+    private List<Course> getCourseList(Request request, Response response) {
+        response.type(CONTENT_TYPE_JSON);
+
+        InputStream is = getClass().getClassLoader().getResourceAsStream("mockCoursesNoTime.json");
+        JsonReader reader = new JsonReader(new InputStreamReader(is));
+
+        List<Course> courses = gson.fromJson(reader, new TypeToken<List<Course>>() {}.getType());
+        return courses;

```

Comment:
Return all course details or just course identifiers?

Reply:
This should only return the data the FE needs to represent courses. That data should be correct in the json file we're using here. Just ID's don't give important things, like name and shifts, that the FE needs from this call.